.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,85 @@
+# contains tests that can run without compiling LightGBM, like
+# linters and documentation checks.
+name: Static Analysis
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+  PYTHON_VERSION: 3.8
+
+jobs:
+  test:
+    name: ${{ matrix.task }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - task: lint
+          - task: check-docs
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: false
+      - name: Setup and run tests
+        shell: bash
+        run: |
+          export TASK="${{ matrix.task }}"
+          export OS_NAME="linux"
```

Comment:
```suggestion
          export OS_NAME="linux"
          COMPILER="gcc"
```

Reply:
why is it necessary to set  a compiler for `check-docs` and `lint`?

Reply:
Although compiler settings are not used directly in `lint` and `check-docs`, a lot of pre-steps rely on `COMPILER` env variable. For example here 
https://github.com/microsoft/LightGBM/blob/415c0cb57794d184ea8639aa5a98a83610639976/.ci/test.sh#L3-L9
and the whole `setup.sh` script. I believe it will be better to explicitly specify compiler to not lose anything during pre-steps in the future that will rely on it.

Reply:
ok sure, no problem